modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/Utils.java

```diff
@@ -208,6 +212,51 @@
         MOSAIC_SUPPORT_FILES_FILTER = initMosaicSupportFilesFilter();
     }
 
+    /** Check if the provided reader is a MultiCRS Reader and it can support the specified crs. */
+    public static boolean isSupportedCRS(GridCoverage2DReader reader, CoordinateReferenceSystem crs)
+            throws FactoryException, IOException {
+        String multiCrsReader =
+                reader.getMetadataValue(AbstractGridCoverage2DReader.MULTICRS_READER);
+        boolean isMultiCRS = multiCrsReader != null && Boolean.valueOf(multiCrsReader);
+        String supportedEpsgCodes = "";
+        if (isMultiCRS) {
+            supportedEpsgCodes =
+                    reader.getMetadataValue(AbstractGridCoverage2DReader.MULTICRS_EPSGCODES);
+        }
+        if (!StringUtils.isBlank(supportedEpsgCodes) && crs != null) {
+            Integer targetCRSEpsgCode = CRS.lookupEpsgCode(crs, true);
```

Comment:
Here is does a full scan... is this intended?

Reply:
Nope. fixing.